components/announcement_bar/cloud_trial_announcement_bar/index.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch} from 'redux';
+
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {GenericAction} from 'mattermost-redux/types/actions';
+import {getStandardAnalytics} from 'mattermost-redux/actions/admin';
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import {getCloudSubscription} from 'mattermost-redux/actions/cloud';
+
+import {getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+
+import {openModal} from 'actions/views/modals';
+
+import {GlobalState} from 'types/store';
+
+import {Preferences} from 'utils/constants';
+
+import {getRemainingDaysFromFutureTimestamp} from 'utils/utils.jsx';
+
+import CloudTrialAnnouncementBar from './cloud_trial_announcement_bar';
+
+function mapStateToProps(state: GlobalState) {
+    const getCategory = makeGetCategory();
+
+    const subscription = state.entities.cloud.subscription;
+    const isCloud = getLicense(state).Cloud === 'true';
+    let isFreeTrial = false;
+    let daysLeftOnTrial = 0;
+
+    if (isCloud && subscription && subscription.trial_end_at > 0 && subscription?.status === 'trialing') {
+        isFreeTrial = true;
+        daysLeftOnTrial = getRemainingDaysFromFutureTimestamp(subscription.end_at);
+    }
```

Comment:
@nickmisasi given that we also need to check payment to determine if you are free trial, should we consider moving this to the subscription model in the server instead?

Reply:
I can add the days left calculation to the backend. 

I don't think we'll need to check the payment method here, it's only that one case via the webhooks where we are hooking into the subscription updated as a substitute for "trial end" event

Reply:
Well if they added their payment during the trial we don't want to show these banners prompting to subscribe anymore